> /Users/runner/work/FreeCAD/FreeCAD/src/Gui/ViewProviderLink.cpp
> Error: /Users/runner/work/FreeCAD/FreeCAD/src/Gui/ViewProviderLink.cpp:2143:12: error: no matching function for call to 'strcmp'
>  2143 |         && strcmp(TypeName, App::PropertyMaterial::getClassTypeId().getName()) == 0) {
>       |            ^~~~~~
> /Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_string.h:89:6: note: candidate function not viable: no known conversion from 'std::string_view' (aka 'basic_string_view<char>') to 'const char *' for 2nd argument
>    89 | int      strcmp(const char *__s1, const char *__s2);
>       |          ^                        ~~~~~~~~~~~~~~~~
> 1 error generated.

Awesome. `string_view`, must be this code-quality-solve-no-bug PR again : https://github.com/FreeCAD/FreeCAD/pull/30008

--- a/src/Gui/ViewProviderLink.cpp
+++ b/src/Gui/ViewProviderLink.cpp
@@ -2116,7 +2116,7 @@ void ViewProviderLink::handleChangedPropertyName(
 {
     // Migration: ShapeMaterial (App::PropertyMaterial) -> ShapeAppearance (App::PropertyMaterialList)
     if (strcmp(PropName, "ShapeMaterial") == 0
-        && TypeName == App::PropertyColor::getClassTypeId().getName()) {
+        && TypeName == App::PropertyMaterial::getClassTypeId().getName()) {
         App::PropertyMaterial prop;
         prop.Restore(reader);
         // Set the whole material object into the list (it will default to index 0)

--- a/src/Gui/ViewProviderLink.cpp
+++ b/src/Gui/ViewProviderLink.cpp
@@ -2116,7 +2116,7 @@ void ViewProviderLink::handleChangedPropertyName(
 {
     // Migration: ShapeMaterial (App::PropertyMaterial) -> ShapeAppearance (App::PropertyMaterialList)
     if (strcmp(PropName, "ShapeMaterial") == 0
-        && TypeName == PropertyColor::getClassTypeId().getName()) {
+        && TypeName == App::PropertyColor::getClassTypeId().getName()) {
         App::PropertyMaterial prop;
         prop.Restore(reader);
         // Set the whole material object into the list (it will default to index 0)

--- a/src/Gui/ViewProviderLink.cpp
+++ b/src/Gui/ViewProviderLink.cpp
@@ -2116,7 +2116,7 @@ void ViewProviderLink::handleChangedPropertyName(
 {
     // Migration: ShapeMaterial (App::PropertyMaterial) -> ShapeAppearance (App::PropertyMaterialList)
     if (strcmp(PropName, "ShapeMaterial") == 0
-        && strcmp(TypeName, App::PropertyMaterial::getClassTypeId().getName()) == 0) {
+        && TypeName == PropertyColor::getClassTypeId().getName()) {
         App::PropertyMaterial prop;
         prop.Restore(reader);
         // Set the whole material object into the list (it will default to index 0)

